app/vmctl/vm_native.go

```diff
@@ -303,3 +312,30 @@ func byteCountSI(b int64) string {
 	return fmt.Sprintf("%.1f %cB",
 		float64(b)/float64(div), "kMGTPE"[exp])
 }
+
+// buildMatchWithFilter adds additional filters from native.Filter.Match to discovered
+// metric names via /api/v1/series API.
+func buildMatchWithFilter(filter string, metricName string) (string, error) {
+	labels, err := promutils.NewLabelsFromString(filter)
+	if err != nil {
+		return "", err
+	}
+
+	var str strings.Builder
+
+	str.WriteString("{")
+	str.WriteString(nameLabel)
+	str.WriteString("=")
+	str.WriteString(strconv.Quote(metricName))
+	for _, label := range labels.Labels {
+		if label.Name == nameLabel {
+			continue
+		}
+		str.WriteString(",")
+		str.WriteString(label.Name)
+		str.WriteString("=")
+		str.WriteString(strconv.Quote(label.Value))
+	}
+	str.WriteString("}")
```

Comment:
Probably, it would be easier to rewrite the code at lines 324-339 into something like the following:

```
labels.Set("__name__", metricName)
return labels.String()
```

Reply:
Used this approach and added a new method, `Set`, with tests for `labels` in the `promutils` library.